File: DecompiledGameScripts/cyberpunk/devices/cleaningMachine/cleaningMachine.swift

public class CleaningMachine extends BasicDistractionDevice {

  protected cb func OnRequestComponents(ri: EntityRequestComponentsInterface) -> Bool {
    super.OnRequestComponents(ri);
  }

  protected cb func OnTakeControl(ri: EntityResolveComponentsInterface) -> Bool {
    super.OnTakeControl(ri);
    this.m_controller = EntityResolveComponentsInterface.GetComponent(ri, n"controller") as CleaningMachineController;
  }

  private const func GetController() -> ref<CleaningMachineController> {
    return this.m_controller as CleaningMachineController;
  }

  public const func GetDevicePS() -> ref<CleaningMachineControllerPS> {
    return this.GetController().GetPS();
  }
}

File: Definitions/Game_Definitions/classes/CleaningMachine.lua
---@meta
---@diagnostic disable

---@class CleaningMachine : BasicDistractionDevice
CleaningMachine = {}

---@return CleaningMachine
function CleaningMachine.new() return end

---@param props table
---@return CleaningMachine
function CleaningMachine.new(props) return end

---@param ri entEntityRequestComponentsInterface
---@return Bool
function CleaningMachine:OnRequestComponents(ri) return end

---@param ri entEntityResolveComponentsInterface
---@return Bool
function CleaningMachine:OnTakeControl(ri) return end

---@return CleaningMachineController
function CleaningMachine:GetController() return end

---@return CleaningMachineControllerPS
function CleaningMachine:GetDevicePS() return end

